data/json/effects.json

```diff
@@ -1495,20 +1525,44 @@
       [ "Your vision improves as your Vitamin A deficiency improves.", "good" ]
     ],
     "max_intensity": 3,
+    "base_mods": { "per_mod": [ -1 ] },
+    "scaling_mods": { "per_mod": [ -2 ] },
     "rating": "bad"
   },
   {
     "type": "effect_type",
     "id": "hypovitB",
-    "name": [ "B12 deficiency", "Slow healing", "No healing" ],
-    "desc": [ "A lack of vitamin B12 in your diet will affect your ability to heal." ],
-    "apply_message": "Simple wounds are starting to concern you more than usual.",
-    "remove_message": "Your wounds now heal normally.",
+    "name": [ "B12 deficiency", "B12 anemia", "B12 dementia" ],
+    "desc": [ 
+      "A lack of vitamin B12 in your diet causes early symptoms of anemia.",
+      "Prolonged lack of vitamin B12 in your diet causes anemia, loss of taste, problems with focus and fading memory.",
+      "Severe vitamin B12 deficiency causes bad case of anemia, loss of taste, and damaged memory, causing dementia.  Your skills rust."
+     ],
+    "apply_message": "Your skin is more pale then usual.",
+    "remove_message": "Your skin tone comes back to its normal color.",
     "decay_messages": [
       [ "Your vitamin B12 deficiency is starting to resolve.", "good" ],
-      [ "Your ability to heal returns as your Vitamin B12 deficiency improves.", "good" ]
+      [ "Your ability to remember things returns as your Vitamin B12 deficiency improves.", "good" ]
     ],
     "max_intensity": 3,
+    "base_mods": {
+      "int_mod": [ -1 ],
+      "dex_mod": [ -1 ],
+      "fatigue_min": [ 5 ],
+      "fatigue_max": [ 10 ],
+      "fatigue_chance": [ 900 ],
+      "h_mod_min": [ -1 ],
+      "h_mod_min_val": [ 0 ],
+      "h_mod_chance": [ 900 ]
+    },
+    "scaling_mods": {
+      "int_mod": [ -2 ],
+      "dex_mod": [ -2 ],
+      "speed_mod": [ -5 ],
+      "fatigue_chance": [ -200 ],
+      "h_mod_min": [ -1 ],
+      "h_mod_chance": [ -200 ]
+    },
```

Comment:
This is pretty reasonable, although the primary effect of b12 deficiency is numbness (and it doesn't always go away). It is very difficult to develop b12 deficiency through dietary insufficiency, but I think that is already modeled by b12 being common in foods in the game.

b12 deficiency also causes anaemia, so you may want to consider copying some of the anaemia effects in here at a lower level.